src/lib.rs

```diff
@@ -744,7 +744,8 @@ fn cstr_to_string(ptr: *const libc::c_char) -> Result<Option<String>, Error> {
     let string = if ptr.is_null() {
         None
     } else {
-        Some(unsafe { CString::from_raw(ptr as _) }.into_string()?)
+        Some(unsafe {
+            CStr::from_ptr(ptr as _) .to_string_lossy().into_owned()})
```

Comment:
Also, clippy is probably warning because you're doing .to_string_lossy() and .into_owned() inside of an unsafe block which isn't necessary.